sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_models.py

```diff
@@ -11,19 +11,19 @@
 
 if TYPE_CHECKING:
     # pylint:disable=unused-import
-    from typing import Any, Dict, Optional
+    from typing import Any, Dict, Optional, Union
     from datetime import datetime
     from ._generated.v7_1 import models as _models
 
 
 class SecretProperties(object):
     """A secret's id and attributes."""
 
-    def __init__(self, attributes, vault_id, **kwargs):
-        # type: (_models.SecretAttributes, str, **Any) -> None
+    def __init__(self, attributes=None, vault_id=None, **kwargs):
```

Comment:
Is there a reason to make these parameters optional?

Reply:
Hm... yeah, that's true, and it doesn't look like we verify anywhere that these properties should be non-`None`. That being said, I'm not sure if we can/should make these property return types optional since these are properties that we should really ensure have a value. @chlowell, is there any context I'm missing for why we assume these properties are populated (aside from things having worked well thus far)?

Reply:
I doubt there's a case in which these have no value--that would be a secret with no ID or management attributes. The autorest client makes these properties optional either by general design or because they aren't required by the spec. I suppose this is one reason to favor instance attributes over properties on models, though that pushes the problem of None checking onto applications.

Reply:
Would it be reasonable to make the property -> instance attribute switch here? If not, I think I'd feel more comfortable ignoring the mypy check here and evaluating this again in a more focused PR/issue

Reply:
I'm uncertain that's the best solution. It doesn't eliminate the need to anticipate None values. And what if there's no case for this library in which SecretBundle.attributes or .id is null? That's a broader question for the service team. For now I think it's reasonable to reflect the reality of the code we have, i.e. to acknowledge that in principle at least, some values could be None, and satisfy mypy by accounting for that. Which is orthogonal to giving these parameters default values--let's not do that. The fact that a parameter could have value None, and so is "optional" in a type checking sense, doesn't imply it should have a default value None, and so be "optional" in a "caller isn't required to pass a value" sense.

Reply:
I'll ignore the mypy warning here and evaluate later.